src/lex.cpp

```diff
@@ -24,35 +41,84 @@ std::vector<token> lex() {
         '*',
         '/'
     };
-    while (std::cin >> i){
-        if (!isspace(i)){
 
-            if (valid.find(i)!=valid.end()){
-                placeholder = i;
-                output.push_back(maketoken(placeholder,line,column));
+    while (std::cin.get(i)) { 
+        //std::cin.get(i) reads input, whitespace, and newlines
+        // https://cplusplus.com/reference/istream/istream/get/
+
+        if (i == '\n') {
+            line++;
+            column = 1;
+            continue;
+        }
+
+        if (!isspace(i)) {
+
+            //if valid operator/parenthesis input, add to output queue:
+            if (valid.find(i) != valid.end()) {
+                lexTokens.push(token(line,column, std::string(1, i)));
             }
 
-            else if(isdigit(i)){
-                placeholder = "/0";
-                bool decimal = false;
-                placeholder += i;
-                while (std::cin >> i && (isdigit(i) || i == '.')){
-                    if (i == '.') {
-                        if (decimal){
-                            //error
+            //if number input or decimal input, check validity:
+            else if (isdigit(i) || i == '.') {
```

Comment:
Don't need to check if the char is '.'
If it is it should go to the else statement that exits the program with an error.
All code related to checking a '.' before the while statement is not needed